iothub/device/tests/Mqtt/MqttIotHubAdapterTest.cs

```diff
@@ -37,6 +39,32 @@ public void TestPopulateMessagePropertiesFromPacket_NormalMessage()
             Assert.AreEqual("MessageId1", message.SystemProperties["message-id"]);
         }
 
+        [TestMethod]
+        [ExpectedException(typeof(MessageTooLargeException), AllowDerivedTypes = false)]
+        public void TestPopulateMessagePropertiesFromMessage_TopicNameTooLong()
+        {
+            var message = new Message();
+
+            for (int i = 0; i < 65; i++)
+            {
+                message.Properties.Add(Guid.NewGuid().ToString(), new string('1', 1024));
+            }
+
+            var publishPacket = new PublishPacket(QualityOfService.AtMostOnce, false, false)
```

Comment:
Are there other things contributing to the length of the topic? (E.g. I see SystemProperties affecting this as well: what are all our the features using SystemProperties?)
/cc @prmathur-microsoft @jebrando @pierreca - please check your SDKs for max topic len as new features add to SystemProperties and app can set a lot of properties as well.